src/chttpd/src/chttpd_db.erl

```diff
@@ -250,20 +251,62 @@ handle_view_cleanup_req(Req, Db) ->
     ok = fabric:cleanup_index_files_all_nodes(Db),
     send_json(Req, 202, {[{ok, true}]}).
 
+handle_partition_design_req(#httpd{
+        path_parts=[_DbName, <<"_partition">>, PartitionKey, _Design, Name, <<"_",_/binary>> = Action | _Rest]
+    }=Req, Db) -> 
+    DbName = mem3:dbname(couch_db:name(Db)),
+    case mem3:is_partitioned(DbName) of
+        false -> throw(<<"Database is not partitioned">>);
```

Comment:
this will be a 500 Server Error? I think you need a different shape to get a 400 Bad Request.